projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -456,15 +457,67 @@ export class IgxGridNavigationService {
         }
     }
 
-    public moveFocusToFilterCell() {
+    public moveFocusToFilterCell(toStart?: boolean) {
         const columns = this.grid.filteringService.unpinnedFilterableColumns;
-        if (this.isColumnFullyVisible(columns.length - 1)) {
-            this.grid.filteringService.focusFilterCellChip(columns[columns.length - 1], false);
+        const targetIndex = toStart ? 0 : columns.length - 1;
+        const isVisible = toStart ? this.isColumnLeftFullyVisible(targetIndex) : this.isColumnFullyVisible(targetIndex);
```

Comment:
`isColumnFullyVisible()` takes visible index and index from the `unpinnedFilterableColumns` collection wouldn't work in some scenarios.